snapcraft/store/_legacy_account.py

```diff
@@ -58,12 +58,16 @@ def _load_potentially_base64_config(config_content: str) -> configparser.ConfigP
     return parser
 
 
-def _get_macaroons_from_conf(conf) -> dict[str, str]:
+def _get_macaroons_from_conf(conf: configparser.ConfigParser) -> dict[str, str]:
     """Format a macaroon and its associated discharge.
 
     :return: A string suitable to use in an Authorization header.
     """
     host = parse_url(os.getenv("UBUNTU_ONE_SSO_URL", constants.UBUNTU_ONE_SSO_URL)).host
+    if not host:
+        raise errors.SnapcraftError(
+            "Couldn't parse environment variable 'UBUNTU_ONE_SSO_URL'."
+        )
```

Comment:
This too seems unrelated to typing.

Reply:
Same as my comment above, the type checkers are now aware this value can be empty, so I need to handle the empty scenario.

In this case, the correct thing to do is raise an error, since this should never be None.